File: backend/src/main/java/com/apocalypse/thefall/dto/item/ItemDto.java
package com.apocalypse.thefall.dto.item;


import com.apocalypse.thefall.entity.item.enums.ItemType;

import java.util.Map;

public interface ItemDto {
    Long id();

    ItemType type();

    Map<String, String> names();

    Map<String, String> descriptions();

    double weight();

    int basePrice();

    String path();
}

File: backend/src/main/java/com/apocalypse/thefall/dto/item/ItemDto.kt
package com.apocalypse.thefall.dto.item

import com.apocalypse.thefall.entity.item.enums.ItemType


interface ItemDto {
    val id: Long?
    val type: ItemType?
    val names: Map<String, String>?
    val descriptions: Map<String, String>?
    val weight: Double?
    val basePrice: Int?
    val path: String?
}
